docs/clients/promtail/stages/tenant.md

```diff
@@ -0,0 +1,75 @@
+# `tenant` stage
+
+The tenant stage is an action stage that sets the tenant ID for the log entry
+picking it from a field in the extracted data map. If the field is missing, the
+default promtail client [`tenant_id`](../configuration.md#client_config) will
+be used.
+
+
+## Schema
+
+```yaml
+tenant:
+  # Name from extracted data to whose value should be set as tenant ID
+  source: <string>
+```
+
+### Example: extract the tenant ID from a structured log
+
+For the given pipeline:
+
+```yaml
+pipeline_stages:
+  - json:
+      expressions:
+        customer_id: customer_id
+  - tenant:
+      source: customer_id
+```
+
+Given the following log line:
+
+```json
+{"customer_id":"1","log":"log message\n","stream":"stderr","time":"2019-04-30T02:12:41.8443515Z"}
+```
+
+The first stage would extract `customer_id` into the extracted map with a value of
+`1`. The tenant stage would set the `X-Scope-OrgID` request header (used by Loki to
+identify the tenant) to the value of the `customer_id` extracted data, which is `1`.
+
+
+### Example: override the tenant ID based on the value of another field
+
+For the given pipeline:
+
+```yaml
+pipeline_stages:
+  - json:
+      expressions:
+        app:
+        message:
+  - labels:
+      app:
+  - match:
+      selector: '{app="api"}'
+      stages:
+        - template:
+            source: tenant_id
+            template: 'team-api'
+        - tenant:
```

Comment:
This is convoluted example with constant tenant ID. If that is a useful use case, maybe tenant stage can support it directly?

Reply:
This originates from a comment made by @cyriltovena, where he was suggesting to add this example too (at least this is what I understood). Cyril, is this example what you meant?

Reply:
Yes because I think that's mainly how people will use it. I don't think someone has a tenant id in its log, or at least this is not the most common.

Reply:
Can the tenant stage support also a const tenant ID ?  I think that's a valid use case.

Reply:
It does not, but it can. What do you think if we add a `value` field, which is mutually exclusive with `source`? For example:

```
# Here the value is hardcoded
- tenant:
    value: team-api

# Here the value is picked from tenant_id field
- tenant:
    source: tenant_id
```

Reply:
@cyriltovena Done! What's your take?